### ISSUE TITLE
compilation error due to stack misalignment

### ISSUE THREAD

**Issue by fedy0** [stnolting/neorv32#907]
**Describe the bug**
When I tried to `make exe` it returns the error shown in the screenshot below

```bash
fedy0@fedy0:~/Documents/repo/neo/neorv32/sw/example/hello_world$ make exe
/opt/riscv32/bin/../lib/gcc/riscv32-unknown-elf/13.2.0/../../../../riscv32-unknown-elf/bin/ld: error: /opt/riscv32/bin/../lib/gcc/riscv32-unknown-elf/13.2.0/../../../../riscv32-unknown-elf/lib/libc.a(libc_a-reent.o) use 4-byte stack aligned but the output use 16-byte stack aligned

```

**To Reproduce**
Steps to reproduce the behavior is [here](https://github.com/fedy0/neo)
It used to work but after I updated my fork of neorv32 it was showing the error below. am i missing something?
It is an alingment issue in the archive libs.

**Expected behavior**
Should compile and execute as [this](https://github.com/fedy0/neo)

**Screenshots**
![image](https://github.com/stnolting/neorv32/assets/26013847/a03db1ca-07dd-4a26-9b2d-2f1b57e86f69)

**Environment:**
 - OS: Ubuntu 22.04.1
 - GCC Version: RISC-V prebuilt toolchain provided by you)
 - Libraries used: newlib

**Hardware:**
 - Hardware version: latest
 - Implemented CPU extensions, peripherals, ...: Default
 - Hardware modifications: None

**Additional context**
None

**Comment by Unike267** [comment]
Hi @fedy0 !

I've tried to reproduce your failure compiling  `hello_world` program with the following container:

- `ghcr.io/stnolting/neorv32/sim `

Using the following command:

```bash
podman run --rm -itv $(pwd):/wrk:Z -w /wrk stnolting/neorv32/sim  make -C sw/example/hello_world/ clean_all info exe
```

*(Here I've used podman but it would be the same for docker.)*

With the repository in the latest version:

6f9abc96dbe1ea57807f26c7878846f0624bb3ea

To see if any of the last pull requests had produced a failure. And I haven't  found any problems.

Everything **works fine.**

I've seen that your submodule points to ed17ae4df64e6a5221562e4adf4de378eaf0c2e8 which is a commit from two months ago. I don't know if that has anything to do with your problem.

Sorry for not been more useful 😅

Cheers! 😃
